colearn_pytorch/pytorch_learner.py

```diff
@@ -90,16 +95,16 @@ def set_weights(self, weights: Weights):
         :param weights: Weights to be stored
         """
 
-        with torch.no_grad():
-            for new_param, old_param in zip(weights.weights,
-                                            self.model.parameters()):
-                old_param.set_(new_param)
+        self.model.load_state_dict(weights.weights)
 
     def train(self):
         """
         Trains the model on the training dataset
         """
 
+        # erase the outdated optimizer memory (momentums mostly)
+        self.optimizer.__setstate__({'state': defaultdict(dict)})
```

Comment:
why are we reseting here? and not in the same function as in tensorflow case?

Reply:
added a separate function & option to reset for the pytorchlearner too. 

We can call the reset either in the `mli_propose_weights` function right before the `self.train()` or in the beginning of the `self.train()`. There is no difference between the two. 

For both Keras and PyTorch I call the reset_optimizer in the beginning of the `self.train()` now in the fix.